src/controller/artifact/processor/processor.go

```diff
@@ -35,12 +37,12 @@ type Addition struct {
 // Processor processes specified artifact
 type Processor interface {
 	// GetArtifactType returns the type of one kind of artifact specified by media type
-	GetArtifactType() string
+	GetArtifactType(ctx context.Context, artifact *artifact.Artifact) string
 	// ListAdditionTypes returns the supported addition types of one kind of artifact specified by media type
-	ListAdditionTypes() []string
+	ListAdditionTypes(ctx context.Context, artifact *artifact.Artifact) []string
```

Comment:
the implemention doesn't consume the `ctx`, do we really need it? Same as ListAdditionTypes

Reply:
In last discussion meeting, we agree to unify params for these four functions. Am I right? @steven-zou